infra/testing/sw-env.js

```diff
@@ -11,21 +11,158 @@
  limitations under the License.
  */
 
-const mockFetch = require('./mock-fetch');
 const {IDBFactory, IDBKeyRange} = require('shelving-mock-indexeddb');
 const makeServiceWorkerEnv = require('service-worker-mock');
+const mockFetch = require('./mock-fetch');
+
+Object.assign(global, makeServiceWorkerEnv());
+global.self = global;
 
 global.fetch = mockFetch;
 global.indexedDB = new IDBFactory();
 global.IDBKeyRange = IDBKeyRange;
-global.importScripts = () => {};
 
-const swEnv = makeServiceWorkerEnv();
+// Stub missing/broken Headers API methods in `service-worker-mock`.
+// https://fetch.spec.whatwg.org/#headers-class
+class Headers {
```

Comment:
Please move all the mocks into a seperate file to clean up this file.

Reply:
BTW, the end goal was to PR this back into `service-worker-mock`.

I can move it, but regardless this was always supposed to be a temporary solution.

Reply:
Done.